### PR TITLE
Segment generate dummy inputs to smaller cases

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

Segment the `generate_dummy_inputs()` method into two cases: PyTorch / TensorFlow for clarity.

### CODE REVIEW

File: optimum/onnx/configuration.py [hunk 1]
```diff
@@ -147,52 +177,21 @@ def generate_dummy_inputs(
...
 
-                        dummy_inputs[label] = tf.zeros(batch_size, dtype=tf.int64)
-        return dummy_inputs
+        if framework == TensorType.PYTORCH and is_torch_available():
+            return self._generate_extra_dummy_inputs_pt(dummy_inputs, label_batch_size, label_seq_length)
+        elif framework == TensorType.TENSORFLOW and is_tf_available():
+            return self._generate_extra_dummy_inputs_tf(dummy_inputs, label_batch_size, label_seq_length)
+        else:
+            raise ValueError(
+                f"Could not generate dummy inputs to perform the ONNX export through {framework} because not installation was found."
```
Comment (echarlaix) [inline optimum/onnx/configuration.py:193]: Something that could be interesting in order to better target the error would be to add some function such [as](https://github.com/huggingface/transformers/blob/main/src/transformers/onnx/features.py#L351)

**Review by echarlaix — APPROVED** [review]
It looks great, thanks @JingyaHuang

File: optimum/onnx/configuration.py (hunk 1 above)
Comment (JingyaHuang) [reply optimum/onnx/configuration.py]: Good point! Thanks Ella, more precise framework information is updated with adding the consideration of frameworks other than PyTorch and TensorFlow. @echarlaix

**Comment by JingyaHuang** [comment]
Clean-up done, will merge.

**Pull request merged**